benchmark_controller.py

```diff
@@ -190,13 +203,14 @@ def main(self):
         #post_build
 
         #pre_run
-        self._run_all(self.benchmark_model.prepare_run_benchmark(self.args.benchmark_run_deps))
+        self._run_all(self.benchmark_model.prepare_run_benchmark(self.args.benchmark_run_deps,
+                                                                 self.compiler_model.getDictCompilers()))
 
         #run
-        stdout, perf_result = self._run_all(self.benchmark_model.run_benchmark(self.binary_name,
+        stdout, perf_result, metadata = self._run_all(self.benchmark_model.run_benchmark(self.binary_name,
                                                                                self.args.benchmark_options), perf=True)
 
-        self._post_run(stdout, perf_result)
+        self._post_run(stdout, perf_result, metadata)
```

Comment:
they seem to be returned all together and used at the same time, so a structure is looking more and more likely a good idea. :)